tests/lib/newlib/newlib_hook/src/main.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <sys/libc-hooks.h>
+
+char buf[8];
+
+/* define the function prototype for sbrk */
+void *sbrk(ptrdiff_t __incr);
+
+/**
+ * @brief Verify _open() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return -1.
+ */
+void test_newlib_stub_open(void)
+{
+	int fd;
+
+	fd = open("../prj.conf", O_RDWR | O_CREAT, 0664);
+	zassert_equal(fd, -1, "open an invalid fd");
+}
+
+/**
+ * @brief Verify _close() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return -1.
+ */
+void test_newlib_stub_close(void)
+{
+	int ret = 0;
+
+	ret = close(-1);
+	zassert_equal(ret, -1, "closed an invalid fd");
+}
+
+/**
+ * @brief Verify _getpid() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_getpid(void)
+{
+	pid_t pid;
+
+	pid = getpid();
+	zassert_equal(pid, 0, "getpid failed");
+}
+
+/**
+ * @brief Verify _fstat() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_fstat(void)
+{
+	struct stat status;
+	int ret;
+
+	ret = fstat(-1, &status);
+	zassert_equal(ret, 0, "fstat failed");
+}
+
+/**
+ * @brief Verify _kill() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_kill(void)
+{
+	int ret;
+
+	ret = raise(SIGILL);
+	zassert_equal(ret, 0, "kill failed");
+}
+
+/**
+ * @brief Verify _read() stub in newlib libc-hooks.c
+ *
+ * @details Use read() functions to test _read().
+ */
+void test_newlib_stub_read(void)
+{
+#ifdef CONFIG_USERSPACE
+	/* Skip userspace because call z_zephyr_read_stdin() failure. */
+	ztest_test_skip();
```

Comment:
#38445 will fix this issue; so once that PR is in, this needs to be updated.

Reply:
yes, after test. read() in userspace is ok now.